fs/opener/ftpfs.py

```diff
@@ -43,8 +43,8 @@ def open_fs(self,
             passwd=parse_result.password,
             proxy=parse_result.params.get('proxy')
         )
-
-        if dir_path:
-            return ftp_fs.opendir(dir_path, factory=ClosingSubFS)
-
-        return ftp_fs
+        return (
+            ftp_fs.opendir(dir_path, factory=ClosingSubFS)
+            if dir_path else
+            ftp_fs
+        )
```

Comment:
Changing this has my `mypy` (v0.590) raise 
```
fs/opener/ftpfs.py:46: error: Incompatible return value type (got "FS", expected "Union[FTPFS, SubFS[FTPFS]]")
```
whereas it did not before (that's why I changed it).

Error does not occur with:
```python
if dir_path:
    return ftp_fs.opendir(dir_path, factory=ClosingSubFS)
else:
    return ftp_fs
```

Reply:
Ah right. I had assumed wrongly that mypy would consider them to be equivalent.